cmd/csi-node-driver-registrar/main.go

```diff
@@ -47,6 +47,7 @@ var (
 	connectionTimeout       = flag.Duration("connection-timeout", 0, "The --connection-timeout flag is deprecated")
 	csiAddress              = flag.String("csi-address", "/run/csi/socket", "Path of the CSI driver socket that the node-driver-registrar will connect to.")
 	kubeletRegistrationPath = flag.String("kubelet-registration-path", "", "Path of the CSI driver socket on the Kubernetes host machine.")
+	livenessProbePort       = flag.Int("liveness-probe-port", 9090, "HTTP port for liveness probe server.")
```

Comment:
CSI drivers on nodes often run with `hostNetwork: true`, which means that the liveness probe port must be open on the node. It becomes increasingly more complicated to make sure the port does not clash with any other port that the node needs.

It should be at least possible to turn the probe off (`--liveness-probe-port=0`?)

Reply:
Maybe don't default the port, and if unset, then don't enable this?

Reply:
That could work. Still, I wish there was an exec probe instead.

Reply:
Can you clarify what you mean by exec probe?

Reply:
I believe they mean doing something like this:
```
livenessProbe:
  exec:
    command:
      - stat
      - /registration/io.kubernetes.storage.mock-reg.sock
```